.github/workflows/build-env-docker.yml

```diff
@@ -0,0 +1,69 @@
+# If this workflow stops working, first consult the documentation page where it was copied from.
+# https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
+
+name: Create and publish a Docker image for bindings build environment
+
+# Configures this workflow to run every time a change is pushed to the branch called `release`.
+on:
+  push:
+    branches: ['main']
```

Comment:
I think with Github's own registry rate limiting is not really an issue, but I will at least limit the concurrency of this job to avoid producing intermediate docker images for commits that aren't top of branch.